services/httpd/handler.go

```diff
@@ -993,8 +1017,437 @@ func fixLiterals(node influxql.Node) {
+			h.httpError(w, fmt.Sprintf("update bucket - %q is not authorized to modify %q: %s", user.ID(), bucket, err.Error()), http.StatusForbidden)
+			return
+		}
+	}
+	if r.ContentLength > 0 {
+		if h.Config.MaxBodySize > 0 && r.ContentLength > int64(h.Config.MaxBodySize) {
+			h.httpError(w, http.StatusText(http.StatusRequestEntityTooLarge), http.StatusRequestEntityTooLarge)
+			return
+		}
+
+		// This will just be an initial hint for the reader, as the
+		// bytes.Buffer will grow as needed when ReadFrom is called
+		bs = make([]byte, 0, r.ContentLength)
+	}
+
+	buf := bytes.NewBuffer(bs)
+	_, err := buf.ReadFrom(r.Body)
+	if err != nil {
+		h.httpError(w, fmt.Sprintf("update bucket - cannot read request body: %s", err.Error()), http.StatusBadRequest)
+		return
+	}
+
+	bu := &BucketUpdate{}
+	if err = json.Unmarshal(buf.Bytes(), bu); err != nil {
+		h.httpError(w, fmt.Sprintf("update bucket - cannot parse request body: %s", err.Error()), http.StatusBadRequest)
+		return
+	}
+	var dur, sgDur time.Duration
+	if len(bu.RetentionRules) > 0 {
+		dur = time.Second * time.Duration(bu.RetentionRules[0].EverySeconds)
+		sgDur = time.Duration(bu.RetentionRules[0].ShardGroupDurationSeconds) * time.Second
+	} else {
+		dur = meta.DefaultRetentionPolicyDuration
+		sgDur = 0
+	}
+	rf := meta.DefaultRetentionPolicyReplicaN
+	m := &meta.RetentionPolicyUpdate{
+		Name:               &bu.Name,
+		Duration:           &dur,
+		ReplicaN:           &rf,
+		ShardGroupDuration: &sgDur,
+	}
+
+	if err := h.MetaClient.UpdateRetentionPolicy(db, rp, m, false); err != nil {
+		h.httpError(w, fmt.Sprintf("update bucket %q: %s", bucket, err.Error()), http.StatusBadRequest)
+		return
+	}
+}
+
+func (h *Handler) serveRetrieveBucketV2(w http.ResponseWriter, r *http.Request, user meta.User) {
+	// This is the API for returning a single bucket
+	// In V1, BucketID and BucketName are the same: "db/rp"
+	db := r.URL.Query().Get(":db")
+	rp := r.URL.Query().Get(":rp")
+	if h.Config.AuthEnabled {
+		if user == nil {
+			h.httpError(w, fmt.Sprintf("retrieve bucket - user is required for database %q", db), http.StatusForbidden)
+			return
+		}
+
+		if err := h.QueryAuthorizer.AuthorizeDatabase(user, influxql.ReadPrivilege, db); err != nil {
+			h.httpError(w, fmt.Sprintf("retrieve bucket - %q is not authorized to read %q: %s", user.ID(), db, err.Error()), http.StatusForbidden)
+			return
+		}
+	}
+
+	if bucket := h.oneBucket(w, fmt.Sprintf("%s/%s", db, rp)); bucket != nil {
+		b, err := json.Marshal(bucket)
+		if err != nil {
+			h.httpError(w, fmt.Sprintf("retrieve bucket marshaling error: %s", err.Error()), http.StatusBadRequest)
+			return
+		}
+		if _, err := w.Write(b); err != nil {
+			h.httpError(w, fmt.Sprintf("retrieve bucket error writing response: %s", err.Error()), http.StatusBadRequest)
+			return
+		}
+	}
+	return
+}
+
+func (h *Handler) serveBucketListV2(w http.ResponseWriter, r *http.Request, user meta.User) {
```

Comment:
nitpick: `RetrieveBucket` above and `BucketList` here are inconsistent, we should attempt to match noun/verb order. Also these are so tied to the http methods that the names might make more sense tied to them as well - `serveGetBuckets`, `serveGetBucketById` ...

Reply:
Fixed across the whole API